### PR TITLE
Fuzz fixes/v3

### PR DESCRIPTION
**Pull request by victorjulien**
Fuzz fixes for smb and stream.

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 10997

File: rust/src/smb/files.rs [hunk 1]
```diff
@@ -89,6 +89,34 @@ impl SMBState {
         return tx_ref.unwrap();
     }
 
+    /// get file tx for a open file. Returns None if a file for the fuid exists,
+    /// but has already been closed.
+    pub fn get_file_tx_by_fuid_with_open_file(&mut self, fuid: &[u8], direction: Direction)
+        -> Option<&mut SMBTransaction>
+    {
+        let f = fuid.to_vec();
+        for tx in &mut self.transactions {
```
Comment (catenacyber) [inline rust/src/smb/files.rs:98]: Do we really need this loop over transactions ?

File: rust/src/smb/smb2.rs [hunk 2]
```diff
@@ -297,7 +297,7 @@ pub fn smb2_write_request_record<'b>(state: &mut SMBState, r: &Smb2Record<'b>)
             };
 
             let mut set_event_fileoverlap = false;
-            let found = match state.get_file_tx_by_fuid(&file_guid, Direction::ToServer) {
```
Comment (catenacyber) [inline rust/src/smb/smb2.rs:300]: Do we still need simple function `get_file_tx_by_fuid` ?

File: src/tests/fuzz/fuzz_sigpcap.c [hunk 3]
```diff
@@ -163,6 +163,7 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
     p->ts.tv_sec = header->ts.tv_sec;
     p->ts.tv_usec = header->ts.tv_usec % 1000000;
     p->datalink = pcap_datalink(pkts);
+    p->pkt_src = PKT_SRC_WIRE;
```
Comment (catenacyber) [inline src/tests/fuzz/fuzz_sigpcap.c:166]: Should this be done for fuzz_predefpcap_aware.c as well ? 
Looks done already for fuzz_sigpcap_aware.c

File: rust/src/smb/smb2.rs (hunk 2 above)
Comment (victorjulien) [reply rust/src/smb/smb2.rs:300]: Yes it's used by the close command and one other case.

File: src/tests/fuzz/fuzz_sigpcap.c (hunk 3 above)
Comment (victorjulien) [reply src/tests/fuzz/fuzz_sigpcap.c:166]: Yeah probably. If I run anything with `--enable-debug` it reaches an assertion otherwise.

Comment (catenacyber) [reply src/tests/fuzz/fuzz_sigpcap.c:166]: Ok, I never ran fuzzing with debug (only debug validation)

Comment (victorjulien) [reply src/tests/fuzz/fuzz_sigpcap.c:166]: It's just when I'm running the reproducer, then I have debug on.

File: rust/src/smb/files.rs (hunk 1 above)
Comment (inliniac) [reply rust/src/smb/files.rs:98]: Goal is to look up a tx in the tx vector, so I think so.

Comment (victorjulien) [reply rust/src/smb/files.rs:98]: > Goal is to look up a tx in the tx vector, so I think so.

Wrong account, wh00ps

**Pull request merged**